modules/gltf/src/lib/extensions/EXT_mesh_features.ts

```diff
@@ -91,3 +100,118 @@ function processMeshPrimitiveFeatures(
+/**
+ * Creates ExtMeshFeatures, creates a featureId containing feature ids provided.
+ * @param scenegraph - Instance of the class for structured access to GLTF data.
+ * @param primitive - target primitive instance that will contain the extension
+ * @param featureIdArray - Array of feature id
+ * @param propertyTableIndex - index of the property table created by the ExtStructuralMetadata (optional).
+ */
+export function createExtMeshFeatures(
+  scenegraph: GLTFScenegraph,
+  primitive: GLTFMeshPrimitive,
+  featureIdArray: NumericArray,
+  propertyTableIndex?: number
+) {
+  if (!primitive.extensions) {
+    primitive.extensions = {};
+  }
+  let extension = primitive.extensions[EXT_MESH_FEATURES_NAME] as GLTF_EXT_mesh_features;
+  if (!extension) {
+    extension = {featureIds: []};
+    primitive.extensions[EXT_MESH_FEATURES_NAME] = extension;
+  }
+
+  const {featureIds} = extension;
+  const featureId: GLTF_EXT_mesh_features_featureId = {
+    featureCount: featureIdArray.length,
+    propertyTable: propertyTableIndex,
+    data: featureIdArray
+  };
+  featureIds.push(featureId);
+
+  scenegraph.addObjectExtension(primitive, EXT_MESH_FEATURES_NAME, extension);
+}
+
+/**
+ * Encodes a feature ID set to extension.
+ * @param scenegraph - Instance of the class for structured access to GLTF data.
+ * @param primitive - Primitive that the data encoded belongs to.
+ * @see https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features
+ */
+function encodeExtMeshFeaturesForPrimitive(
+  scenegraph: GLTFScenegraph,
+  primitive: GLTFMeshPrimitive
+) {
+  const extension = primitive.extensions?.[EXT_MESH_FEATURES_NAME] as GLTF_EXT_mesh_features;
+  if (!extension) {
+    return;
+  }
+  const featureIds: GLTF_EXT_mesh_features_featureId[] = extension.featureIds;
+  featureIds.forEach((featureId, ind) => {
+    if (featureId.data) {
+      const {accessorKey, index} = createAccessorKey(primitive.attributes);
+      const typedArray = new Uint32Array(featureId.data as NumericArray);
+
+      // Clean up featureId object.
+      // Everything that could come from the original extension in case of round-trip decode/encode operations should be deleted.
+      // We need make sure the featureId object is clean.
+      featureIds[ind] = {
+        featureCount: typedArray.length,
+        propertyTable: featureId.propertyTable,
+        attribute: index
+      };
+
+      scenegraph.gltf.buffers.push({
+        arrayBuffer: typedArray.buffer,
+        byteOffset: typedArray.byteOffset,
+        byteLength: typedArray.byteLength
+      });
+
+      const bufferViewIndex = scenegraph.addBufferView(typedArray);
+      const accessorIndex = scenegraph.addAccessor(bufferViewIndex, {
+        size: 1,
+        componentType: getComponentTypeFromArray(typedArray),
+        count: typedArray.length
+      });
+      primitive.attributes[accessorKey] = accessorIndex;
+    }
+  });
+}
+
+function createAccessorKey(attributes: {}) {
```

Comment:
Please add documentation and types to every function

Reply:
Done.